src/olcs/core/olimageryprovider.js

```diff
@@ -247,23 +252,46 @@ olcs.core.OLImageryProvider.prototype.getTileCredits = function(x, y, level) {
  * @override
  */
 olcs.core.OLImageryProvider.prototype.requestImage = function(x, y, level) {
-  const tileUrlFunction = this.source_.getTileUrlFunction();
-  if (tileUrlFunction && this.projection_) {
-
-    // Perform mapping of Cesium tile coordinates to OpenLayers tile coordinates:
-    // 1) Cesium zoom level 0 is OpenLayers zoom level 1 for EPSG:4326
-    const z_ = this.tilingScheme_ instanceof Cesium.GeographicTilingScheme ? level + 1 : level;
-    // 2) OpenLayers tile coordinates increase from bottom to top
-    const y_ = -y - 1;
-
-    let url = tileUrlFunction.call(this.source_,
-        [z_, x, y_], 1, this.projection_);
-    if (this.proxy_) {
-      url = this.proxy_.getURL(url);
-    }
-    return url ? Cesium.ImageryProvider.loadImage(this, url) : this.emptyCanvas_;
+  // Perform mapping of Cesium tile coordinates to ol3 tile coordinates:
+  // 1) Cesium zoom level 0 is OpenLayers zoom level 1 for EPSG:4326
+  const z_ = this.tilingScheme_ instanceof Cesium.GeographicTilingScheme ? level + 1 : level;
+  // 2) OpenLayers tile coordinates increase from bottom to top
+  const y_ = -y - 1;
+
+  const tilegrid = this.source_.getTileGridForProjection(this.projection_);
+  if (z_ < tilegrid.getMinZoom() || z_ > tilegrid.getMaxZoom()) {
+    return Promise.resolve(this.emptyCanvas_); // no data
+  }
+
+  const tile = this.source_.getTile(z_, x, y_, 1, this.projection_);
+
+  tile.load();
+
+  // not yet loaded!
+  // const image = tile.getImage();
+  // if (!image || !image.src) {
+  //   return Promise.resolve(this.emptyCanvas_); // no data
+  // }
+
+
+  const state = tile.getState();
+  if (state === ol.TileState.LOADED || state === ol.TileState.EMPTY) {
+    return tile.getImage() || undefined;
```

Comment:
This return also needs a promise wrapper if `tile.getImage()` is not undefined

Reply:
Thanks, done.